### ISSUE TITLE
Define the next release (0.11)

### ISSUE THREAD

**Issue by zathras-crypto** [OmniLayer/omnicore#363]
Hi all,

There seems to be some disparity around expectations on what is going to make it into our next release (0.11).

The release will be based on Bitcoin Core 0.10 using our 0.0.10 Omni Core branch as a base - this much is clear.

The following is a list of items in no particular order that various people have requested make it into the next release and their current status.  I'm sure I'm also missing some.  Let's please use this issue to lock in explicitly what we are going to put into 0.11:

Consensus critical requests needing feature activation:
- Trading of non-Omni pairs on the MetaDEx   [coding 100% complete, currently in testing]
- Fee caching/distribution system & MetaDEx fees   [coding 99% complete, currently in testing]
- Allow any property to use DEx phase 1   [coding 0% complete, not currently started]
- Allow BTC in property crowdsales   [coding 0% complete, not currently started]

Non consensus critical requests:
- Provide trading of non-Omni pairs in the UI   [coding 0% complete, not currently started]
- Provide support in the UI for DEx phase 1   [coding 50% complete, work-in-progress stalled]
- Provide support for property issuance in the UI   [coding 0% complete, not currently started]

Note:  Just because they're listed here doesn't mean I'm suggesting they should be included in 0.11 - I'm just listing those requests I'm aware of.

Note:  it's assumed that everything we've built & merged already for what was going to be our 0.0.10.1 release (so all the bugfixes, stuff like payload creation on RPC etc) will all be included in 0.11 instead.

Please post your feedback and thoughts about exactly what should be included in 0.11.

Thanks all
Z

**Comment by CraigSellars** [comment]
> Allow any property to use DEx phase 1 [coding 0% complete, not currently started]
I’m not in favor of this - I like the idea of OMNI being the way “into” the system.  I’d be ok with a multi-step process that the client handles to get people into OMNI then place trades on the Phase II DEx once the Phase I tx is done… :-)

> Allow BTC in property crowdsales [coding 0% complete, not currently started]
while this is a very very nice to have, I don’t think it should make it into 0.11.

>Provide trading of non-Omni pairs in the UI [coding 0% complete, not currently started]
>Provide support in the UI for DEx phase 1 [coding 50% complete, work-in-progress stalled]
>Provide support for property issuance in the UI [coding 0% complete, not currently started]
I am in favor of these features as a priority, but I think only the first one should be included in 0.11 if possible (simple drop-downs for each column).

I’d like for us to get 0.11 with non-Omni pairs live as quickly as possible, and I’d be fine with a 0.11.1 with the additional UI pieces.

**Comment by zathras-crypto** [comment]
OK, so the general feedback I've gathered so far seems to be:

**0.11** (release candidate ASAP)
* Trading of non-Omni pairs on the MetaDEx [coding 100% complete, currently in testing]
* Fee caching/distribution system & MetaDEx fees [coding 99% complete, currently in testing]

**0.11.1** (can be released when it's ready before or after activating 0.11 features as not consensus critical)
* Provide trading of non-Omni pairs in the UI [coding 0% complete, not currently started]
* Provide support in the UI for DEx phase 1 [coding 50% complete, work-in-progress stalled]
* Provide support for property issuance in the UI [coding 0% complete, not currently started]

**Deferred** (will not be included in the 0.11 branch but can be added to next major release - 0.12 - if the consensus is that these features are a good idea)
* Allow any property to use DEx phase 1 [coding 0% complete, not currently started]
* Allow BTC in property crowdsales [coding 0% complete, not currently started]

Could you please signal if you agree with this approach @dexX7 @CraigSellars @patrickdugan @msgilligan @achamely 

Thanks
Z

**Comment by dexX7** [comment]
I'm fine with the MetaDEx changes, but hesitated regarding the UI extensions, given the context of #362.

**Comment by zathras-crypto** [comment]
Yeah I can understand that.  FYI I got the general feeling that it would be desirable to get more DEx capability into the current QT UI, following which people would be open to tabling alternatives.  Would that be your take from the feedback supplied so far also?

So would you agree for 0.11, it's just the UI stuff for 0.11.1 you're not sure about?

Do you have an alternate path you'd prefer dude?  Really my key goal is to get a roadmap set, even if it is just for this next release - something to guide the work - I kind of feel (at least from my own perspective and contributions) that it's very scattered and I find myself doing lots of different things and trialling/prototyping and talking to lots of people and so forth but not really making huge progress from an Omni Core PoV.

In a nutshell I'm just really keen to lock in direction :)

**Comment by dexX7** [comment]
> In a nutshell I'm just really keen to lock in direction :)

Thanks for pushing this!

> Would that be your take from the feedback supplied so far also?

My understanding was that Sean and Patrick want a DEx v1 integration into the UI, and the general sentiment was "having an Omni UI would be nice". I share the sentiment, but I'm not sure, if it's the best route forward, given the small end user base.

> So would you agree for 0.11, it's just the UI stuff for 0.11.1 you're not sure about?

Correct. I'm fine with the consensus changes for 0.11, but split regarding the UI stuff.

> Do you have an alternate path you'd prefer dude?

Moving on top of 0.12 is also still an open topic, which could be tackled after the soon-ish 0.10 release.

**Comment by zathras-crypto** [comment]
> My understanding was that Sean and Patrick want a DEx v1 integration into the UI, and the general sentiment was "having an Omni UI would be nice"

Craig was also keen on non-Omni pair trading in the UI - from a personal view I'd be more interested in this than I would be in DEx v1 integration, simply because we're releasing *that* feature now, and I'd rather provide UI support for a feature we're activating now to help it gain traction as opposed to a feature we activated a couple of years ago which is already rarely used.

I'm wondering if I should start in on MetaDEx UI changes in QT while you guys running the OmniJ/Jenkins/Unit Test stuff are working up the testing side of the consensus changes for 0.11.

Alternatively I could use this as an opportunity to get more involved in the testing side of things myself perhaps.

> Moving on top of 0.12 is also still an open topic, which could be tackled after the soon-ish 0.10 release.

I'm very keen on moving to 0.12.  The kind of deadlines being requested preclude doing that for 0.11 but I'm definitely supportive of moving to a 0.12 base ASAP after that.  If for nothing else than the substantial reduction in compute (and thus cost) required for users/integrators to run a node (with the move to libsecp256k1).

0.12 also allows pruning with a wallet, which is of interest.  We can't use it directly as we're still required to load blocks in some places, but I wonder whether we can be selective and exclude Omni-containing blocks from the pruning process.  Disk consumption is one of the pain points to using Omni Core (though this is dwarfed by the startup sync time as the most vocal objection).

**Comment by patrickdugan** [comment]
Hey, given time constraints, I'm inclined to defer all Omnicore UI updates possibly to 0.12.1 and get key features done sooner rather than later. 

BTC for crowdsales is nice but I'm about to start doing PARK as a managed smart property, and can receive BTC as a matter of app logic. 

So, let's get 0.11 out ASAP and start on 0.12, and budget these nice to have details for later.

**Comment by zathras-crypto** [comment]
> Hey, given time constraints, I'm inclined to defer all Omnicore UI updates possibly to 0.12.1 and get key features done sooner rather than later.

To re-iterate my concerns from the meeting, are we not just making exactly the same mistakes over and over and not learning anything?

We continuously build features, and then don't provide any mechanism for people to use them.  The features then go essentially unused.  To back up this assertion you only have to look at:
* How many people have actually used STO since we built it and made it live?
* How many people have actually used Send-All?
* How many people have actually made a trade on the MetaDEx?

On and on it goes, and the answer to all of those questions is essentially zero.

Thus I'm extremely hesitant to continue down this path and once again say "let's not worry about access to our features & instead let's build another new feature (smart contracts) and continue to ignore the fact we have an unsustainably low number of users".

I know this isn't a popular viewpoint, but I care about this project (I've put 3 years of my life into it) and I am just quite disappointed to see us repeatedly doing the same thing that hasn't worked for us in the past.

From my own perspective, one key thing above all else is actually getting users - the ability to exchange value - that's at the core of everything - exchanging value - whether that be sending tokens or trading tokens.  That's what's going to drive "users" and get us the critical mass we need to sustain the project long enough to do advanced stuff like smart contracts properly (ie with a white paper, spec and proper design and implementation).

Thus I really, really wish we could actually focus on the user experience.  The UX for sending tokens is reasonable, and that's why over 99% of our transactions are sending tokens.  The UX for trading tokens is terrible (and non-existent outside Omni Core) and that's why ~0.001% of our transactions are trades.  Other features we spent months building in the past don't even get used at all (ie 0%).

TL:DR; my vote is absolutely to focus on the UX for sending and trading - the bread and butter if you will - in the hopes of bringing some fresh life into the project.

I will go with the consensus as I'm a team player, but I implore consideration and learning from the past before making decisions.  Another bunch of time of building stuff no-one uses I'm sure will hurt the project :(

**Comment by zathras-crypto** [comment]
Note, tone is hard to convey in text - I'm not tring to be standoff-ish guys :) I'm just sharing an opinion which you may or may not want to take on board :)

**Comment by dexX7** [comment]
@zathras-crypto: I generally share your opinion.

Do you have something particular in mind how to improve the UX? And are you referring to Omni Core, Omni Wallet or a new on-ramp for users?

**Comment by zathras-crypto** [comment]
> @zathras-crypto: I generally share your opinion.

That's good & thanks dude, I'm glad I'm not the only one who worries about this stuff :)

> Do you have something particular in mind how to improve the UX? And are you referring to Omni Core, Omni Wallet or a new on-ramp for users?

I think what we need to do is be realistic and target our efforts towards the audiences that we want.  If I may ramble for a moment...

**Omni Core**
Our audience here is quite clearly enthusiasts, developers and integrators.  We don't have very many end users on Omni Core because of barriers to entry (like sync times and disk consumption).  I see this platform as our primary and most critical stack, but I also acknowledge the reality on the ground is that end users don't like it and we haven't been successful growing a strong user base with it.

**Omni Wallet**
This is where I think most non-technical users end up. From a UX PoV I think it's probably the best that we have so far.  But trading is going to be a slow burn to get there I think (just from the vibes I've picked up).  Also on security it's great that it does client side signing but let's be honest, no-one verifies the transactions that are given to the client to sign.  This is not Omni Wallet's fault but also a reality on the ground.  A compromised Omni Wallet server could easily craft the transaction maliciously saying "Send John Doe all the Bitcoins at this address" and an Omni Wallet user would sign it client side just the same.  Thus I think Omni Wallet is great and accessible and I'd love to see trading brought online, but at the end of the day it's still a "web wallet" and a web wallet can't be our only offering to end users.

**Exchanges**
From various impressions & discussions I've had I think a not insignificant number of users don't use an Omni capable wallet *at all*.  They buy/sell Omni properties to trade with on exchanges but only withdraw/deposit BTC.  Other folks simply deposit & withdraw Omni properties directly from one exchange to another and don't use their own Omni capable wallet.

**OmniPortfolio**
If this is something that we can realistically release as a productized wallet in the near term I'm super positive about it.  I haven't been following the development in detail so I'm not sure where it's up to right now.

**New on-ramp**
This.  Ideally, we'd build a product targeted for end users making sure to address the problems they currently face such as ease of access and availability.  I actually think it's not as big an undertaking as it sounds and could be done relatively quickly if we focused, especially if we fork one of the many existing wallet projects, and doubly-so if this idea of using several API services in tandem to ensure consensus is acceptable (I'm of split minds on that).

Of course I don't have a full roadmap or anything, as at the moment I'm just trying to secure agreement on a direction.  Once we have a direction, we can plot specific steps.

Maybe I'm just making myself unpopular with our leadership haha, but I'm an opinionated dude so what can you do :)

**Comment by dexX7** [comment]
While there are some very good points in this thread, I currently don't see any actionable items we can extract here, so I'm going to close it for now. Please feel free to reopen though. :)
